packages/clerk-js/src/ui/components/UserProfile/DeleteUserForm.tsx

```diff
@@ -37,13 +38,16 @@ export const DeleteUserForm = withCardStateProvider((props: DeleteUserFormProps)
         throw Error('user is not defined');
       }
 
+      // Store the session so it can be ended after the user is deleted
+      const session = client.lastActiveSessionId;
       await user.delete();
-
       // TODO: Investigate if we need to call `setActive` with {session: null}
-      if (otherSessions.length === 0) {
-        return navigateAfterSignOut();
-      }
-      await navigateAfterMultiSessionSingleSignOutUrl();
+      const navigationCallback =
+        otherSessions.length === 0 ? navigateAfterSignOut : navigateAfterMultiSessionSingleSignOutUrl;
+      return await setActive({
+        session,
```

Comment:
⛔ We should not use the previous session since it's (probably) already deleted. We should pass `{ session: null }`